### PR TITLE
Fix first purchase estimate calculation

### LINKED ISSUE #43: New items should be categorized based on user input
Clarification from team meeting: new items should be categorized based on user input (soon, kind of soon, etc)

- [ ] update Daria and Abby's feature #11  from this past week to initialize the next purchase date estimate with the user input estimate

_Originally posted by @cafloyd in https://github.com/the-collab-lab/tcl-22-smart-shopping-list/issues/13#issuecomment-835562151_

### PR DESCRIPTION
**Pull request by dariatsvetkova**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

[Previous fix](https://github.com/the-collab-lab/tcl-22-smart-shopping-list/pull/44) is not working as expected because the sorting logic is based on previous purchase dates.

This PR adds a new field `dateAdded` to the database to store the date when the user added the item and set its first estimated purchase date. This date is then used to put the item into one of the list segments based on how many days are left until its first estimated purchase.

## Related Issue

Closes #43
(for good)

## Acceptance Criteria

* When a new item is added, it shows up in [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/)
* The new item in Firebase has the field `purchaseEstimates` containing 1 element that is equal to the frequency set by the user in the Add Groceries form and the field `dateAdded` that contains the current date.
* The new item is put into the correct segment of the grocery list.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |

### Before

https://www.loom.com/share/39708b3a417847b4aaff28961684ba02

### After

https://www.loom.com/share/e8751fb42b7c4b3a82e6ed7dc606f373

## Testing Steps / QA Criteria

To test the feature, use one of the following ways:

1. Follow the [preview link](https://tcl-22-shopping-list--pr58-dt-new-item-estimate-zbiqo5e2.web.app)
2. Launch the app locally from your terminal:
* Pull down this branch with `git pull origin dt-new-item-estimate-fix` and check that branch out with `git checkout origin dt-new-item-estimate-fix`
* Run `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

When on application:

* In Add Items view, add a new item and notice the frequency you are setting it to.
* In [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/data), your item's `purchaseEstimates` should have the value that you set and its `dateAdded` should be the current date.
* The item should appear in the correct segment of the grocery list based on the value you set.

### CODE REVIEW

File: src/Item.js [hunk 1]
```diff
@@ -50,24 +57,26 @@ function Item({ userToken, item, status }) {
       const newDate = new Date();
       const newDates = [...purchaseDates, newDate];
       const numberOfPurchases = newDates.length;
-      let lastEstimate;
-      let latestInterval;
+      // let latestInterval;
```
Comment (laurenmbeatty) [inline src/Item.js:60]:
```suggestion
```

File: src/Item.js [hunk 2]
```diff
@@ -34,13 +40,14 @@ function Item({ userToken, item, status }) {
...
-  }, [purchaseDates]);
+  }, [purchaseDates, dateAdded]);
 
   const handleClick = () => {
     if (checked) {
       // Remove the latest purchase date and estimate from the database:
       purchaseDates.pop();
-      purchaseEstimates.length > 0 && purchaseEstimates.pop();
+      // Don't remove the first estimate that comes from user input:
+      purchaseEstimates.length > 1 && purchaseEstimates.pop();
```
Comment (laurenmbeatty) [inline src/Item.js:50]: 👍

**Review by ablizben — APPROVED** [review]
Thanks for this, Daria! This looks great.

**Review by laurenmbeatty — APPROVED** [review]
LGTM 👍

File: src/ItemList.js [hunk 3]
```diff
@@ -30,22 +30,26 @@ function ItemList(props) {
 
     // Separate list items into categories based on how soon they should be bought
     props.list.forEach((item) => {
-      if (
-        item.purchaseDates.length < 1 ||
-        !(item.purchaseEstimates?.length > 0)
-      ) {
+      if (!(item.purchaseEstimates?.length > 0)) {
```
Comment (cafloyd) [inline src/ItemList.js:33]: nice use of optional chaining!

File: src/ItemList.js [hunk 4]
```diff
@@ -30,22 +30,26 @@ function ItemList(props) {
...
-        );
+        // If the item hasn't been bought yet,
+        // calculate lastInterval based on the date when it was added to the list
+        // and how soon the user wanted to buy it:
+        const lastPurchase =
+          item.purchaseDates.length > 0
+            ? fromUnixTime(
+                item.purchaseDates[item.purchaseDates.length - 1].seconds,
+              )
+            : fromUnixTime(item.dateAdded?.seconds);
```
Comment (cafloyd) [inline src/ItemList.js:44]: nice use of optional chaining!

**Review by cafloyd — APPROVED** [review]

File: src/ItemList.js [hunk 5]
```diff
@@ -30,22 +30,26 @@ function ItemList(props) {
...
-      ) {
+      if (!(item.purchaseEstimates?.length > 0)) {
         resultsObj['inactive'].push(item);
       } else {
-        const lastPurchase = fromUnixTime(
-          item.purchaseDates[item.purchaseDates.length - 1].seconds,
-        );
+        // If the item hasn't been bought yet,
+        // calculate lastInterval based on the date when it was added to the list
+        // and how soon the user wanted to buy it:
```
Comment (kkatoch89) [inline src/ItemList.js:38]: Nice. I like this logic

**Review by kkatoch89 — APPROVED** [review]
Looks good to me! You're on fire @dariatsvetkova 🔥🔥🔥

**Pull request merged**